tests/reg_issue764/reg_issue764.fz

```diff
@@ -40,4 +40,4 @@ reg_issue764 is
   say (add_ints [1, 2, 3, 4, 5])
   say (add_ints (array 0 i->i))
   say (add_strings ["hello", "world"])
-  say (add_strings (array 0 i->" "))
+  say (add_strings (array 0 (i -> $" ")))
```

Comment:
I was confused here, would be good to add a description comment at the beginning of this test (before `reg_issue764 is`) explaining what this does. 

Does this work after your patch: 
```
  count(elemens Sequence T) =>
    for 
       res := 0, res+1
       _ in elems
     else
       res
```
?
This was at least my original intention, even though I am not sure that this is necessarily a good idea.

Reply:
Yes, above code works after patch.